eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -37,19 +37,32 @@ jobs:
           pip install doc-warden
           ward scan -d $(Build.SourcesDirectory) -c $(Build.SourcesDirectory)/eng/.docsettings.yml
         displayName: "Verify Readmes"
-      - task: DownloadPipelineArtifact@1
+      - task: DownloadPipelineArtifact@2
         displayName: "Download Build Artifacts"
-        condition: and(succeededOrFailed(), ne(variables['Build.Reason'],'PullRequest'))
+        condition: succeededOrFailed()
         inputs:
-          artifactName: packages
-          targetPath: $(System.DefaultWorkingDirectory)
+          artifact: packages
+          path: $(Pipeline.Workspace)/packages
```

Comment:
Why use `$(Pipeline.Workspace)` over `$(System.DefaultWorkingDirectory)`?

Reply:
Based on the guidance here: https://docs.microsoft.com/en-us/azure/devops/pipelines/artifacts/pipeline-artifacts?view=azure-devops&tabs=yaml#downloading-artifacts

> By default, the **Download Pipeline Artifact** task downloads files to `$(Pipeline.Workspace)`. This is the default and recommended path for all types of artifacts.

Reply:
Also I discovered that the documentation appears to be incorrect regarding the behavior of the `path` parameter. If a single artifact is downloaded, the ***contents*** of the artifact are downloaded to whatever path you provide (or the default, `$(Pipeline.Workspace)`, if no path is provided). No enclosing directory is created, which is why I had to include the artifact name as part of the `path`. This directly contradicts what the documentation says:

> Unless the specified download path is absolute, a folder with the same name as the artifact is created under the download path, and the artifact's files are placed in it.
>  …
> ```yaml
> steps:
> - download: current
>   artifact: WebApp
>   patterns: **/*.js
> ```
> Files (with the directory structure of the artifact preserved) are downloaded under $(Pipeline.Workspace)/WebApp.


Just to add more confusion, the example in the task’s documentation shows the default value to be:
> `#path: '$(System.ArtifactsDirectory)'`

which is clearly wrong and probably was missed when they updated the docs from V1 to V2 of the task.

Reply:
Thanks for the info.